### DISCUSSION TITLE
Is there a qutebrowser.el?

### DISCUSSION THREAD

**Discussion by teledyn** [qutebrowser/qutebrowser#8178] [General]
Given the efficient simplicity, and the key bindings, someone somewhere must have implemented `browse-url-browser-function` in elisp by now, right? This would be perfect for `browse-url-secondary-browser-function` for those problem pages.

**Comment by The-Compiler** [comment]
Converting this to a discussion as there doesn't seem anything actionable for qutebrowser itself here.

**Comment by teledyn** [comment]
I will give no warrantees other than to say it works for at least two of my purposes, as shift-F9 and as the '&' option in EWW, but just to start things off, here's my ugly hack of the firefox equivalent code:
```
  ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
  ;; qutebrowser?
  ;; mostly works; shamelessly lifted from the firefox code
  ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
  (setq browse-url-qutebrowser-program "qutebrowser"
		  browse-url-secondary-browser-function 'browse-url-qutebrowser
		  browse-url-qutebrowser-arguments "")
	(global-set-key (kbd "S-<f9>") 'browse-url-qutebrowser)

	(defun browse-url-qutebrowser (url &optional new-window)
		  "Ask the qutebrowser to load URL.
		Defaults to the URL around or before point.  Passes the strings
		in the variable `browse-url-qutebrowser-arguments' to Qutebrowser.

		Interactively, if the variable `browse-url-new-window-flag' is non-nil,
		loads the document in a new Qutebrowser window.  A non-nil prefix argument
		reverses the effect of `browse-url-new-window-flag'.

		If `browse-url-qutebrowser-new-window-is-tab' is non-nil, then
		whenever a document would otherwise be loaded in a new window, it
		is loaded in a new tab in an existing window instead.

		Non-interactively, this uses the optional second argument NEW-WINDOW
		instead of `browse-url-new-window-flag'."
		  (interactive (browse-url-interactive-arg "URL: "))
		  (setq url (browse-url-encode-url url))
		  (let* ((process-environment (browse-url-process-environment)))
			(apply #'start-process
				   (concat "qutebrowser " url) nil
				   browse-url-qutebrowser-program
				   (append
					browse-url-qutebrowser-arguments
					(if (browse-url-maybe-new-window new-window)
				(if browse-url-qutebrowser-new-window-is-tab
					# this is not meant to work just copied from firefox
					'("-new-tab")
				  '("-new-window")))
					(list url)))))

	(defun org-open-at-point-with-qutebrowser ()
	  (interactive)
	  (let ((browse-url-browser-function 'browse-url-qutebrowser))
	  (org-open-at-point )))
  
```
As noted it is missing equivalents to those last two options, I'm not sure they are even important or when that might be called. Oh and apologies for the leading spaces, it was in an orgmode block formatted for lisp and I wasn't sure how to just copy the code so I used ctrl-space and pasted what I got, it's late, everyone has vi and can deal with it, I'm sure.

**Reply by The-Compiler** [reply]
I don't know anything about emacs, but you might be looking for `--target tab` and `--target window` for those arguments.

**Reply by teledyn** [reply]
I know precious little about qutebrowser, so we're even! Thanks for that tip, I'll fold it into mine and see what happens!

**Comment by lrustand** [comment]
There is now! Check out my new package here: https://github.com/lrustand/qutebrowser.el
